init.lua

```diff
@@ -0,0 +1,572 @@
+map("n", "x", "\"_x", "Deletes a single character without yanking it.")
+
+map("n", "j", "gj", "`j` but it respects soft line wraps.")
+map("n", "k", "gk", "`k` but it respects soft line wraps.")
+
+-- Yanking and pasting using the system clipboard.
+--
+-- If you're using WSL make sure your Windows clipboard is actually accessible from within WSL.
+-- See `:help clipboard-wsl` for more information.
+--
+-- If you're on Linux make sure to install either `xclip` or `wl-clipboard` (depending on whether
+-- you use X11 or Wayland).
+
+map({ "n", "v" }, "<Leader>y", "\"+y", "Yanks to the system clipboard.")
+map({ "n", "v" }, "<Leader>p", "\"+p", "Pastes from the system clipboard.")
+map({ "n", "v" }, "<Leader>P", "\"+P", "Pastes from the system clipboard without yanking the deleted text.")
+
+-- Stay in visual mode when indenting / dedenting text
+map("x", ">", ">gv", "Indents text in visual mode without losing visual selection.")
+map("x", "<", "<gv", "Dedents text in visual mode without losing visual selection.")
+
+-- Move lines around
+map("n", "<A-j>", ":m .+1<CR>==", "Moves the current line down by 1.")
+map("n", "<A-k>", ":m .-2<CR>==", "Moves the current line up by 1.")
+map("x", "<A-j>", ":m '>+1<CR>=gv", "Moves the current line down by 1.")
+map("x", "<A-k>", ":m '<-2<CR>=gv", "Moves the current line up by 1.")
+
+-- Navigate various lists
+map("n", "[q", ":cprev<CR>", "Jumps to the previous entry in the quickfix list.")
+map("n", "]q", ":cnext<CR>", "Jumps to the next entry in the quickfix list.")
+map("n", "[Q", ":cfirst<CR>", "Jumps to the first entry in the quickfix list.")
+map("n", "]Q", ":clast<CR>", "Jumps to the last entry in the quickfix list.")
+
+map("n", "[b", ":bprev<CR>", "Jumps to the previous buffer.")
+map("n", "]b", ":bnext<CR>", "Jumps to the next buffer.")
+
+map("n", "[d", vim.diagnostic.goto_prev, "Jumps back one diagnostic message.")
+map("n", "]d", vim.diagnostic.goto_next, "Jumps forward one diagnostic message.")
+map("n", "gl", vim.diagnostic.open_float, "Opens a floating window with diagnostics from the current line.")
+
+-- Navigate between windows
+map({ "n", "v" }, "<C-h>", "<cmd>wincmd h<CR>", "Moves 1 window left.")
+map({ "n", "v" }, "<C-j>", "<cmd>wincmd j<CR>", "Moves 1 window down.")
+map({ "n", "v" }, "<C-k>", "<cmd>wincmd k<CR>", "Moves 1 window up.")
+map({ "n", "v" }, "<C-l>", "<cmd>wincmd l<CR>", "Moves 1 window right.")
+
+-- Better defaults for terminal mode
+map("t", "<C-]>", "<C-\\><C-n>", "Goes from terminal mode to normal mode.")
+map("t", "<S-Space>", "<Space>")
+map("t", "<C-Space>", "<Space>")
+map("t", "<S-BS>", "<BS>")
+map("t", "<C-BS>", "<C-w>")
+map("t", "<C-CR>", "<CR>")
+
+-- Execute the current config file.
+--
+-- You can use this to reload your config while you're editing it without having to restart nvim.
+map("n", "<Leader>x", function()
+  if vim.o.filetype == "vim" or vim.o.filetype == "lua" then
+    vim.cmd("write | source %")
```

Comment:
```suggestion
    vim.cmd.source()
```

Reply:
Does sourcing automatically write the file as well? That was kind of the intention.

Reply:
I am aware. But that's not the intention. It is supposed to save and execute, not just execute. Maybe wasn't obvious from the comment, but this is intentional.

Reply:
I see, yeah didn't think that was the intention